social_protection/models.py

```diff
@@ -57,6 +58,18 @@ def clean(self):
 
     def __str__(self):
         return f'{self.individual.first_name} {self.individual.last_name}'
+    
+    @classmethod
+    def get_queryset(cls, queryset, user):
+        if queryset is None:
+            queryset = cls.objects.all()
+
+        individuals = Individual.objects.filter(
+            id__in=queryset.values('individual_id')
+        ).distinct()
+    
+        individual_queryset = Individual.get_queryset(individuals, user)
+        return queryset.filter(individual__in=individual_queryset)
```

Comment:
may be a subquery would be more efficient, did you check what this code generate as SQL queries ?